### ISSUE TITLE
flutter_compass-Swift.h file not found

### ISSUE THREAD

**Issue by cikenstrip** [hemanthrajv/flutter_compass#3]
Have been playing with the plugin for couple of month and it did great on Android devices! But when I'm using and compiling into my iPhone (real device, not emulator), build failed.

```
Starting Xcode build...                                          
Xcode build done.                                           22.1s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    /Users/umroh/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_compass-0.0.
    2/ios/Classes/FlutterCompassPlugin.m:2:9: fatal error:
    'flutter_compass/flutter_compass-Swift.h' file not found
    #import <flutter_compass/flutter_compass-Swift.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.

Could not build the precompiled application for the device.
```

Is there something (Pod configuration or anything) that I missed? Thanks.

**Comment by hemanthrajv** [comment]
If your ios project uses objective-c, please try enabling swift support.

**Comment by itoon** [comment]
I have a same issue. It have after set podfile config.build_settings['SWIFT_VERSION'] = '4.2'

**Comment by itoon** [comment]
@cikenstrip I've resolved it already.
First, you have to add config.build_settings['SWIFT_VERSION'] = '4.2' like this
`config.build_settings['ENABLE_BITCODE'] = 'NO' `\
`config.build_settings['SWIFT_VERSION'] = '4.2'`
and add use_frameworks!
`target 'Runner' do`\
`use_frameworks!`

and in iOS you have to add key Privacy - Location When In Use Usage Description and Privacy - Location Always and When In Use Usage Description in info.plist

Good luck

**Comment by itoon** [comment]
> Have been playing with the plugin for couple of month and it did great on Android devices! But when I'm using and compiling into my iPhone (real device, not emulator), build failed.
> 
> ```
> Starting Xcode build...                                          
> Xcode build done.                                           22.1s
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     ** BUILD FAILED **
> 
> Xcode's output:
> ↳
>     /Users/umroh/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_compass-0.0.
>     2/ios/Classes/FlutterCompassPlugin.m:2:9: fatal error:
>     'flutter_compass/flutter_compass-Swift.h' file not found
>     #import <flutter_compass/flutter_compass-Swift.h>
>             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     1 error generated.
> 
> Could not build the precompiled application for the device.
> ```
> Is there something (Pod configuration or anything) that I missed? Thanks.

@cikenstrip I've resolved it already.
First, you have to add config.build_settings['SWIFT_VERSION'] = '4.2' like this
config.build_settings['ENABLE_BITCODE'] = 'NO' 
config.build_settings['SWIFT_VERSION'] = '4.2'
and add use_frameworks!
target 'Runner' do
use_frameworks!

and in iOS you have to add key Privacy - Location When In Use Usage Description and Privacy - Location Always and When In Use Usage Description in info.plist

Good luck

**Comment by hemanthrajv** [comment]
Hi @itoon , please try with the latest version.

**Comment by nanunh** [comment]
I ran into the exact same error today when I tried to run the vanilla example file from 
https://pub.dartlang.org/packages/flutter_compass#-example-tab-

Overcame the issue by making all of the suggested changes - 
1. Edited Podfile from Visual Studio Code Version 1.32.1 - i.e added the four lines - 
```
target 'Runner' do
  use_frameworks!

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
```
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['SWIFT_VERSION'] = '4.2'

```  
2. Had to make two  changes to pubspec.yaml 
2a.  added dependencies for
` flutter_compass: ^0.2.0`
2b. added   assets:
`    - assets/compass.jpg`

3. added the imports shown below to the file @ https://pub.dartlang.org/packages/flutter_compass#-example-tab-
```
import 'package:flutter/material.dart';
import 'package:flutter_compass/flutter_compass.dart';
import 'dart:math' as math;

```
The error then went away when I ran the Compass example from menu item "Start Without Debugging".  
Thx!
